### PR TITLE
https-ify some links

### PR DESCRIPTION
**Pull request by asottile**
apologies in advance for the size of this.  I didn't even finish getting through all the files :(

I started by opening all files that had `http://` in my editor -- got through about ~50 files before realizing there were 400+ files to edit and then moved to a `sed` approach instead for ones I knew were https

NO ENTRY

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env python
 """Distutils based setup script for SymPy.
 
-This uses Distutils (http://python.org/sigs/distutils-sig/) the standard
+This uses Distutils (https://python.org/sigs/distutils-sig/) the standard
 python mechanism for installing packages. Optionally, you can use
-Setuptools (http://pythonhosted.org/setuptools/setuptools.html)
+Setuptools (https://pythonhosted.org/setuptools/setuptools.html)
```
Comment (asmeurer) [inline setup.py]: This link is dead.

File: sympy/functions/special/error_functions.py [hunk 2]
```diff
@@ -2142,7 +2142,7 @@ class fresnels(FresnelIntegral):
     References
     ==========
 
-    .. [1] http://en.wikipedia.org/wiki/Fresnel_integral
+    .. [1] https://en.wikipedia.org/wiki/Fresnel_integral
     .. [2] http://dlmf.nist.gov/7
```
Comment (asmeurer) [inline sympy/functions/special/error_functions.py]: This URL can be fixed.

File: sympy/ntheory/egyptian_fraction.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def egyptian_fraction(r, algorithm="Greedy"):
     References
     ==========
 
-    .. [1] http://en.wikipedia.org/wiki/Egyptian_fraction
+    .. [1] https://en.wikipedia.org/wiki/Egyptian_fraction
     .. [2] https://en.wikipedia.org/wiki/Greedy_algorithm_for_Egyptian_fractions
     .. [3] http://www.ics.uci.edu/~eppstein/numth/egypt/conflict.html
```
Comment (asmeurer) [inline sympy/ntheory/egyptian_fraction.py]: This URL can be fixed.

**Comment by asottile** [comment]
@asmeurer note that some of the files I just ran `sed` over and did not audit every link -- if you find ones that can be fixed please just commit to my branch <3

**Comment by asmeurer** [comment]
I checked all the domains I wasn't sure about, and everything seems to work (except the one I commented on). I also checked a handful of links that I noticed while going through, but not all of them.

**Comment by asottile** [comment]
:thinking: I'm bad at rst -- any idea what this is on about?

```
/home/travis/build/sympy/sympy/doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst:5:Duplicate explicit target name: "solvers.".
```

File: doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst [hunk 4]
```diff
@@ -309,8 +309,8 @@ Mathematical Equivalents
 | solve for.            |                       | refer to              |
 |                       |                       | ``nonlinsolve`` and   |
 |                       |                       | ``nsolve`` in         |
-|                       |                       | `solvers. <http://docs|
-|                       |                       | .sympy.org/latest/    |
+|                       |                       | `solvers. <https://   |
+|                       |                       | docs.sympy.org/latest/|
```
Comment (asmeurer) [inline doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst]: I guess it's referring to this. Sphinx doesn't like it when you have two links to the same URL with different names for some reason. 

Actually, this link is rather silly. It should be a relative link, not a link to docs.sympy.org. It looks like this needs to be fixed throughout this document (and possibly elsewhere in the docs as well).

Comment (asmeurer) [reply doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst]: Opened https://github.com/sympy/sympy/issues/15380 for the docs.sympy.org links. If you don't want to fix the other links for this PR that's fine, just this one link needs to be made relative to fix the build.

Comment (asottile) [reply doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst]: ok, just to clarify this should link to `/latest/modules/solvers/solvers.html` or is there an `rst` way to link to the specific document?

Comment (asmeurer) [reply doc/src/modules/physics/mechanics/sympy_mechanics_for_autolev_users.rst]: There's a way to cross reference the document directly. You might have to add a label to the top of solvers.rst.

**Comment by asottile** [comment]
if I never have to rewrite another rst table it'll be too soon

**Comment by asmeurer** [comment]
I always use emacs's table mode that makes it easier to edit ascii tables, but even then, it's annoying.

File: README.rst [hunk 5]
```diff
@@ -49,14 +49,14 @@ To get the git version do
     $ git clone git://github.com/sympy/sympy.git
 
 For other options (tarballs, debs, etc.), see
-http://docs.sympy.org/dev/install.html.
+https://docs.sympy.org/dev/install.html.
```
Comment (vorabrijesh) [inline README.rst]: Should this become :ref:`install <install>`
This is regarding issue #15380 in which we should use relative links.
If so I will open a PR.

**Pull request merged**